File: backend/app/models/memory.py
from datetime import datetime
from sqlalchemy import String, Text, Integer, DateTime, JSON, Boolean
from sqlalchemy.orm import Mapped, mapped_column
from app.core.database import Base


class TravelMemory(Base):
    __tablename__ = "travel_memories"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    session_id: Mapped[str] = mapped_column(String(100), nullable=False, index=True)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    original_content: Mapped[str] = mapped_column(Text, nullable=False)
    polished_content: Mapped[str | None] = mapped_column(Text)
    spot_name: Mapped[str | None] = mapped_column(String(200))
    spot_id: Mapped[str | None] = mapped_column(String(50))
    source_type: Mapped[str] = mapped_column(String(20), nullable=False, default="chat")  # chat / vision / route
    mood_tag: Mapped[str | None] = mapped_column(String(20))  # 敬畏/惊喜/平静/感动
    metadata_json: Mapped[dict | None] = mapped_column(JSON)
    # 记忆胶囊字段
    is_capsule: Mapped[bool] = mapped_column(Boolean, default=False)
    capsule_unlock_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    capsule_content: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)


class JourneySummary(Base):
    __tablename__ = "journey_summaries"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    session_id: Mapped[str] = mapped_column(String(100), nullable=False, index=True)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    spot_count: Mapped[int] = mapped_column(Integer, default=0)
    memory_count: Mapped[int] = mapped_column(Integer, default=0)
    date_range: Mapped[str] = mapped_column(String(100))
    cover_image_url: Mapped[str | None] = mapped_column(String(500))
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

File: frontend/src/api/memory.ts
import request from './request';

export interface TravelMemory {
  id: number;
  session_id: string;
  title: string;
  original_content: string;
  polished_content: string | null;
  spot_name: string | null;
  spot_id: string | null;
  source_type: string;
  mood_tag: string | null;
  metadata_json: Record<string, any> | null;
  created_at: string;
  updated_at: string;
}

export interface JourneySummary {
  id: number;
  session_id: string;
  title: string;
  content: string;
  spot_count: number;
  memory_count: number;
  date_range: string;
  cover_image_url: string | null;
  created_at: string;
}

export interface GenerateResponse {
  new_count: number;
  total_count: number;
  memories: TravelMemory[];
}

export async function listMemories(sessionId: string): Promise<TravelMemory[]> {
  const resp = await request.get<TravelMemory[]>('/memory/list', {
    params: { session_id: sessionId },
  });
  return resp.data;
}

export async function generateMemories(sessionId: string): Promise<GenerateResponse> {
  const resp = await request.post<GenerateResponse>('/memory/generate', {
    session_id: sessionId,
  });
  return resp.data;
}

export async function polishMemory(memoryId: number): Promise<TravelMemory> {
  const resp = await request.post<TravelMemory>(`/memory/${memoryId}/polish`);
  return resp.data;
}

export async function generateSummary(sessionId: string): Promise<JourneySummary> {
  const resp = await request.post<JourneySummary>('/memory/summary/generate', {
    session_id: sessionId,
  });
  return resp.data;
}

export async function getLatestSummary(sessionId: string): Promise<JourneySummary | null> {
  const resp = await request.get<JourneySummary | null>('/memory/summary/latest', {
    params: { session_id: sessionId },
  });
  return resp.data;
}
